javaparser-symbol-solver-testing/src/test/java/com/github/javaparser/symbolsolver/Issue1814.java

```diff
@@ -0,0 +1,88 @@
+package com.github.javaparser.symbolsolver;
+
+import com.github.javaparser.*;
+import com.github.javaparser.ast.CompilationUnit;
+import com.github.javaparser.ast.Modifier;
+import com.github.javaparser.ast.body.ClassOrInterfaceDeclaration;
+import com.github.javaparser.ast.body.MethodDeclaration;
+import com.github.javaparser.ast.expr.SimpleName;
+import com.github.javaparser.ast.stmt.BlockStmt;
+import com.github.javaparser.ast.type.ClassOrInterfaceType;
+import com.github.javaparser.resolution.declarations.ResolvedMethodDeclaration;
+import com.github.javaparser.resolution.declarations.ResolvedReferenceTypeDeclaration;
+import com.github.javaparser.symbolsolver.javaparsermodel.declarations.JavaParserClassDeclaration;
+import com.github.javaparser.symbolsolver.model.resolution.SymbolReference;
+import com.github.javaparser.symbolsolver.model.resolution.TypeSolver;
+import com.github.javaparser.symbolsolver.resolution.AbstractResolutionTest;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.List;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+/**
+ * @author Dominik Hardtke
+ * @since 01.09.2018
+ */
+public class Issue1814 extends AbstractResolutionTest {
+    private JavaParser javaParser;
+
+    @Before
+    public void setup() {
+        // construct a fake java.lang.Object class with only one method (java.lang.Object#equals(java.lang.Object)
+        ClassOrInterfaceDeclaration fakeObject = new ClassOrInterfaceDeclaration();
+        fakeObject.setName(new SimpleName("java.lang.Object"));
```

Comment:
I think SimpleName should not contain dots. You may want to set a package declaration instead.

Reply:
How would I do that? I didn't find a way to associate a `PackageDeclaration` with my `ClassOrInterfaceDeclaration`

Reply:
You create a CompilationUnit then you set the PackageDeclaration in the CompilationUnit and insert the ClassOrInterfaceDeclaration in the same CompilationUnit. When we want to find the qualified name of a class we do that by looking in which CompilationUnit is contained and reading the associated package declaration